app/backend/src/core/controllers/appraiser/handlers/BlockRewardsHandler.ts

```diff
@@ -14,31 +14,45 @@ export interface BlockRewardSummary {
 }
 
 export class BlockRewardsHandler {
-  private provider: IBlockchainClient;
   private priceService: PriceService;
 
-  constructor(provider: IBlockchainClient, priceService: PriceService) {
-    this.provider = provider;
+  constructor(priceService: PriceService) {
     this.priceService = priceService;
   }
 
-  async handleBlockRewards(block: Block): Promise<BlockRewardSummary> {
+  async handleBlockRewards(
+    block: Block,
+    blockTransactionReceipts: TransactionReceipt[] | undefined,
+  ): Promise<BlockRewardSummary> {
     let totalGasFees = BigInt(0);
     let totalTips = BigInt(0);
 
-    for (const tx of block.transactions) {
-      const receipt = await this.provider.getTransactionReceipt(tx.hash);
+    if (!blockTransactionReceipts) {
+      return {
+        gasFees: totalGasFees,
+        gasFeesUsd: 0,
```

Comment:
Was a bit misleading for me - to set 0 to USD fields but use variables for non-USD values. Maybe we should set 0 to all to make it clearer on why USD conversion is skipped in this case?